### ISSUE TITLE
Crash AWSS3TransferUtility.m line 1451

### ISSUE THREAD

**Issue by pablogeek** [aws-amplify/aws-sdk-ios#1232]
**Describe the bug**
I got this crash from Fabric 

AWSS3TransferUtility.m line 1451
Fatal Exception: NSInvalidArgumentException
Cannot read file at file:///var/mobile/Containers/Data/Application/CDD5C0C8-4F6A-4CBD-B901-1D13ABFBC14E/Library/Caches/S3TransferUtility/102A076D-946B-443E-B8B2-A45D0D2D5ED1

**To Reproduce**
It is not reproducible in my devices

**Which AWS service(s) are affected?**
S3
**Expected behavior**
Not crash

**Environment(please complete the following information):**
 - SDK Version: [e.g. 2.6.29] 2.8.4
 - Dependency Manager: Cocoapods
 - Swift Version : 4.2

**Device Information (please complete the following information):**
 Ipad mini 2, iOS 12

**Comment by kvasukib** [comment]
@pablogeek Sorry for the inconvenience caused. As I can see this error occurred because the SDK couldn't read the file from the path constructed.

```
NSURLSessionUploadTask *nsURLUploadTask = [self->_session uploadTaskWithRequest:urlRequest
                                                                         fromFile:[NSURL fileURLWithPath:subTask.file]];
```

`fileURLWithPath` initializes and returns a newly created NSURL object as a file URL with a specified path. The SDK uses the file name you specify when uploading data through `AWSS3TransferUtility`. Can you give an example of the file name you pass in to the `uploadData` method?

**Comment by pablogeek** [comment]
Hello @kvasukib 
this is an example of our urls 
`file:///var/mobile/Containers/Data/Application/7C049BE9-B1A9-4D42-83C0-9DECB04AF3F2/Documents/E4A4E65F-DC32-4F01-B658-5EA366441448.m4v`

Thanks

**Comment by kvasukib** [comment]
@pablogeek Thank you for the quick response. What is the name of the key that is passed in to `uploadData`?

```swift
transferUtility.uploadData(data,
       bucket: "YourBucket",
       key: "YourFileName",
       contentType: "text/plain",
       expression: expression,
       completionHandler: completionHandler).continueWith {
          (task) -> AnyObject! in
              if let error = task.error {
                 print("Error: \(error.localizedDescription)")
              }

              if let _ = task.result {
                 // Do something with uploadTask.
              }
              return nil;
      }
```

**Comment by kvasukib** [comment]
@pablogeek Are you still facing the issue?

**Comment by pablogeek** [comment]
@kvasukib I had only a couple of this. So seems it is not happening anymore.

**Comment by kvasukib** [comment]
@pablogeek Thank you for your quick response. Since it's intermittent and not reproducible from our end based on the key name, I'm closing the issue. Please feel free to re-open when you encounter the issue again with the key-name passed into `uploadData`.

**Comment by pablogeek** [comment]
Fatal Exception: NSInvalidArgumentException
Cannot read file at file:///var/mobile/Containers/Data/Application/E2B57380-C330-4B2D-90F3-D077F2B05378/Library/Caches/S3TransferUtility/267BCC43-1E9B-428C-8C12-345BB964BB95

@kvasukib  it is happening again for a user :(

**Comment by AliJ91** [comment]
Any update on this issue? im facing it also

**Comment by ruisebas** [comment]
Hi @AliJ91. The underlying cause for this particular crash was fixed in [2.12.1](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.12.1).

If you are still getting a crash using the latest SDK release, please open a new issue and share as much information as possible, such as:
- Full symbolicated crashlog.
- Steps to reproduce.
- File name and service key used.
- Any relevant code snippets.

Thanks!
